test/functional/commands/general/mobile-command-e2e-specs.js

```diff
@@ -6,19 +6,47 @@ import { initDriver } from '../../helpers/session';
 chai.should();
 chai.use(chaiAsPromised);
 
-describe('mobile: shell', function () {
+describe('mobile', function () {
   let driver;
   before(async function () {
-    driver = await initDriver(APIDEMOS_CAPS);
+    delete APIDEMOS_CAPS.app;
+    driver = await initDriver({
+      ...APIDEMOS_CAPS,
+
+      // For deeplinking to work, it has to run a session in a native
+      // context but it doesn't matter what native app is run so just
+      // run io.appium.settings for simplicity
+      appPackage: 'io.appium.settings',
+      appActivity: '.Settings',
+    });
   });
   after(async function () {
     await driver.quit();
   });
-  it('should call execute command without proxy error, but require relaxed security flag', async function () {
-    try {
-      await driver.execute('mobile: shell', {command: 'echo', args: ['hello']});
-    } catch (e) {
-      e.message.should.match(/Original error: Appium server must have relaxed security flag set in order to run any shell commands/);
-    }
+  describe('mobile:shell', function () {
+    it('should call execute command without proxy error, but require relaxed security flag', async function () {
+      try {
+        await driver.execute('mobile: shell', {command: 'echo', args: ['hello']});
+      } catch (e) {
+        e.message.should.match(/Original error: Appium server must have relaxed security flag set in order to run any shell commands/);
+      }
+    });
+  });
+  describe('mobile:deepLink', function () {
+    it('should be able to launch apps using Instant Apps', async function () {
+      try {
+        await driver.execute("mobile: deepLink", {url: 'https://www.realtor.com/realestateandhomes-search/San-Jose_CA', package: 'com.move.realtor'});
+
+        // Check that the source has the package name somewhere
+        await driver.source().should.eventually.match(/com\.move\.realtor/);
+
+        // Check that we can find a native element and interact with it
+        const btn = await driver.elementsByXPath('//android.widget.Button');
+        btn.length.should.be.above(0);
+        await btn[0].click();
+      } catch (e) {
+        e.message.should.match(/unable to resolve intent/i);
```

Comment:
i don't think a test should have two possible different verification scenarios. either it should test the success or it should test the failure deterministically. here it's testing one or the other and we'll never know which.

Reply:
Yeah I had the same thought, I"m just not sure how to go about determining if the Android Emulator that we're testing on supports Instant Apps.

Reply:
@jlipps I just changed it so that _only_ execute is inside the try-catch block. It occurred to me that having assertions inside of a try-catch block is probably bad practice.